src/R/Components/Impl/ConnectionManager/Implementation/Connection.cs

```diff
@@ -20,5 +30,43 @@ internal class Connection : ConnectionInfo, IConnection {
         /// If true, the connection is to a remote machine
         /// </summary>
         public bool IsRemote { get; }
+
+        public override string UserProvidedPath {
+            get { return base.UserProvidedPath; }
+            set {
+                base.UserProvidedPath = value;
+                Path = ToCompletePath(value);
+            }
+        }
+
+        public static string ToCompletePath(string path) {
```

Comment:
I don't think we should store this `short` path at all. It is just a shortcut that we provide during the input - similar to tab in VS code editor. Since the full path contains the whole short path, we don't limit user in editing it lately. But at least he will edit the real path that is used for connection.